src/repr/src/row.rs

```diff
@@ -419,70 +416,92 @@ where
 {
     match tag {
         Tag::BytesTiny | Tag::StringTiny => {
-            push_copy!(data, bytes.len() as u8, u8);
+            let len = bytes.len() as u8;
+            data.push(len);
         }
         Tag::BytesShort | Tag::StringShort => {
-            push_copy!(data, bytes.len() as u16, u16);
+            let len = bytes.len() as u16;
+            data.extend_from_slice(&len.to_le_bytes());
         }
         Tag::BytesLong | Tag::StringLong => {
-            push_copy!(data, bytes.len() as u32, u32);
+            let len = bytes.len() as u32;
+            data.extend_from_slice(&len.to_le_bytes());
         }
         Tag::BytesHuge | Tag::StringHuge => {
-            push_copy!(data, bytes.len() as usize, usize);
+            let len = u64::cast_from(bytes.len());
+            data.extend_from_slice(&len.to_le_bytes());
         }
         _ => unreachable!(),
     }
     data.extend_from_slice(bytes);
 }
 
+fn push_date<D>(data: &mut D, date: NaiveDate)
+where
+    D: Vector<u8>,
+{
+    data.extend_from_slice(&date.year().to_le_bytes());
```

Comment:
For this (and `push_time`) we should probably extend the vector once up front rather than possibly doing two allocations and copies in the pathological case

Reply:
Actually now I’m finding plenty of cases where we do a series of pushes without an extend first so no need to fix this one case of that now, we can follow up with that later if we think it matters.